ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java

```diff
@@ -570,13 +572,84 @@ public void putErrorCasesTest() {
+
+    assertTrue("Should be true if either of accountId and containerId is unknown",
+        BlobStore.validateAuthorization(Utils.getRandomShort(TestUtils.RANDOM), Container.UNKNOWN_CONTAINER_ID,
+            Utils.getRandomShort(TestUtils.RANDOM), Utils.getRandomShort(TestUtils.RANDOM)));
+
+    short accountIdInStore = Utils.getRandomShort(TestUtils.RANDOM);
+    short containerIdInStore = Utils.getRandomShort(TestUtils.RANDOM);
+    assertFalse("Return false if both mismatch.",
+        BlobStore.validateAuthorization(accountIdInStore, containerIdInStore, Utils.getRandomShort(TestUtils.RANDOM),
+            Utils.getRandomShort(TestUtils.RANDOM)));
+    assertTrue("Should return if both match.",
+        BlobStore.validateAuthorization(accountIdInStore, containerIdInStore, accountIdInStore, containerIdInStore));
+    assertFalse("Return false if accountId is mismatch",
+        BlobStore.validateAuthorization(accountIdInStore, containerIdInStore, Utils.getRandomShort(TestUtils.RANDOM),
+            containerIdInStore));
+    assertFalse("Return false if containerId is mismatch",
+        BlobStore.validateAuthorization(accountIdInStore, containerIdInStore, accountIdInStore,
+            Utils.getRandomShort(TestUtils.RANDOM)));
+  }
+
+  /**
+   * Test DELETE authorization failure for {@link BlobStore#delete(MessageWriteSet)}.
+   * Success is covered by other tests.
+   */
+  @Test
+  public void deleteAuthorizationFailureTest() throws Exception {
+    MockId key = deletedKeys.iterator().next();
+    if (store.getStoreConfig().storeValidateAuthorization == true) {
+      MessageInfo info = new MessageInfo(key, DELETE_RECORD_SIZE, (short) 0, (short) 0, System.currentTimeMillis());
+      MessageWriteSet writeSet =
+          new MockMessageWriteSet(Collections.singletonList(info), Collections.singletonList(ByteBuffer.allocate(1)));
+      try {
+        store.delete(writeSet);
+        fail("Store DELETE should have failed");
+      } catch (StoreException e) {
+        assertEquals("Unexpected StoreErrorCode", StoreErrorCodes.Authorization_Failure, e.getErrorCode());
+      }
+    }
+    verifyDeleteFailure(key, StoreErrorCodes.ID_Deleted);
+  }
+
+  /**
+   * Test GET authorization failure for {@link BlobStore#get(List, EnumSet)}.
+   * Success is covered by other tests.
+   */
+  @Test
+  public void getAuthorizationFailureTest() throws Exception {
+    MockId addedId = put(1, PUT_RECORD_SIZE, Utils.Infinite_Time).get(0);
+    assertNotSame("accountId should not be -1 for added Blob.", -1, addedId.getAccountId());
+    assertNotSame("containerId should not be -1 for added Blob.", -1, addedId.getContainerId());
+    if (store.getStoreConfig().storeValidateAuthorization == true) {
+      verifyGetFailure(new MockId(addedId.getID()), StoreErrorCodes.Authorization_Failure);
+    } else {
+      store.get(Collections.singletonList(addedId), EnumSet.noneOf(StoreGetOptions.class));
```

Comment:
why do this at all? To make sure it succeeds? Shouldn't we check that the return is ok in that case?